helper/call.go

```diff
@@ -705,21 +706,17 @@ func (c *ContractCaller) CurrentStateCounter(stateSenderInstance *statesender.St
 	return result
 }
 
-// CheckIfBlocksExist - check if latest block number is greater than end block
+// CheckIfBlocksExist - check if the given block exists on local chain
 func (c *ContractCaller) CheckIfBlocksExist(end uint64) bool {
-	// Get Latest block number.
-	var latestBlock *ethTypes.Header
+	// Get block by number.
+	var block *ethTypes.Header
 
-	err := c.MaticChainRPC.Call(&latestBlock, "eth_getBlockByNumber", "latest", false)
+	err := c.MaticChainRPC.Call(&block, "eth_getBlockByNumber", fmt.Sprintf("0x%x", end), false)
```

Comment:
Aren't we changing this to get header by block number?

Reply:
Ignore this, just read the description in the PR.

Reply:
Yes, But plan changed after testing, The issue was something else.